### ISSUE TITLE
`db:schema:load` does not restore the ActiveRecord::Base connection after running

### ISSUE THREAD

**Issue by Taytay** [rails/rails#42912]
I believe I've stumbled over a bug in the new rake tasks that allow for multi-database handling.

Some of the tasks seem to clean up after themselves by re-establishing the connection for ActiveRecord::Base after they are done:
https://github.com/rails/rails/blob/d9fda555942bfb6bcb1f75477b1425e2ab3162f5/activerecord/lib/active_record/railties/databases.rake#L106-L108

However, the database-specific call to `load` (`db:schema:load:some_db_name`) does not re-establish a connection for ActiveRecord::Base in the rake file or in the underlying DatabaseTasks file:
https://github.com/rails/rails/blob/d9fda555942bfb6bcb1f75477b1425e2ab3162f5/activerecord/lib/active_record/railties/databases.rake#L477-L480

This means that if you have a rake task (like we do) that invokes this task manually, it leaves ActiveRecord::Base connected to whatever database you happened to last call `db:schema:load` for.

We have a rake task that invokes other rake tasks, and that's how we ran into it.
In short, we are doing something like:
```ruby
# rake task preamble stuff...
    Rake::Task['db:schema:load:non_primary_db'].invoke
    # At this point, ActiveRecord::Base is still connected to that non-primary DB.
    # Any calls we make to other rake tasks at this point that make use of `ActiveRecord::Base.connection` are going to get
    # connected to the non_primary_db!

    # So, this task might fail if it is relying upon the expected Rails behavior of respecting the ApplicationRecord's database configuration:
    Rake::Task['some:custom:task'].invoke
# ...
```

My expectation/preference would be that any rake task run would be expected to _always_ clean up after itself if it ever modifies the ActiveRecord::Base. Is that a reasonable expectation, or am I doing something wrong?

**Comment by ghiculescu** [comment]
> My expectation/preference would be that any rake task run would be expected to _always_ clean up after itself if it ever modifies the ActiveRecord::Base. Is that a reasonable expectation, or am I doing something wrong?

Yes, I think you're right. Would you be keen to make a PR?

**Comment by intrip** [comment]
@Taytay please let us know if you're interested in opening a PR otherwise I'll be happy to start working on this.

**Comment by intrip** [comment]
@ghiculescu https://github.com/rails/rails/pull/43050 should fix this.

**Comment by Taytay** [comment]
Thanks @intrip and sorry for my silence. I didn't have time to set up my contributing environment and I appreciate you taking this one!
